### PR TITLE
OSD-10025: Disable PagerDuty service when a cluster is in limited support

### PR DESCRIPTION
**Pull request by supreeth7**
Resolves: [OSD-10025](https://issues.redhat.com/browse/OSD-10025)
<hr>

- This PR switches the PagerDuty service enabled/disabled based on the `api.openshift.com/limited-support` label in the cluster deployment.
- When `api.openshift.com/limited-support` is set to **true** the cluster related PagerDuty service is disabled and any active alerts are automatically resolved.
- When `api.openshift.com/limited-support` is set to anything else other than true or if the label is deleted the cluster related PagerDuty service is re-enabled and alerting is resumed.

### CODE REVIEW

File: pkg/controller/pagerdutyintegration/clusterdeployment_limited_support.go
Comment (jewzaam) [inline pkg/controller/pagerdutyintegration/clusterdeployment_limited_support.go:42]:
```suggestion
	} else if ok {
```

If the label is not set it should be treated as not(limited-support) always.  If the value is anything other than `"true"` also treat as not(limited-support).

**Review by jewzaam — CHANGES_REQUESTED** [review]
/hold

Open question generally in JIRA on why this is in code VS using PDI selector.  If that conversation is resolved and we want to do this disable then please remove hold (aka don't wait for me).

**Review by jewzaam — CHANGES_REQUESTED** [review]
Conclusion from conversations is we can update the "osd" PDI and delete the secret when limited support label is present.  Addons do not depend on the "osd" PDI's PD secret.

Please update to move this into the "osd" PDI configuration.  Thanks!

**Comment by supreeth7** [comment]
Thank you for looking into this and figuring out the ideal approach!
I will be closing this PR in favor of #175.

**Comment by supreeth7** [comment]
After the recent discussions, conclusion; PD service history matters to SRE hence deleting the service isn't a good idea. It is ideal to go with this (enable/disable PD service) approach instead. And it further ensures consistency with the existing hibernation flow.

File: pkg/controller/pagerdutyintegration/clusterdeployment_limited_support.go
Comment (wshearn) [inline pkg/controller/pagerdutyintegration/clusterdeployment_limited_support.go:42]: What happens if the label is deleted instead of being set to false? Wouldn't the cluster stay in limited support?

Comment (supreeth7) [reply pkg/controller/pagerdutyintegration/clusterdeployment_limited_support.go]: @wshearn thank you for pointing this out. I assumed the label will always exist with a value, this has now been fixed in a way that prevents redundant calls to PD API to check if a service has been disabled each the label doesn't exist.

**Comment by supreeth7** [comment]
@jewzaam can you please give this a one final look and approve if everything looks good?

File: pkg/controller/pagerdutyintegration/clusterdeployment_limited_support.go
Comment (jewzaam) [inline pkg/controller/pagerdutyintegration/clusterdeployment_limited_support.go:61]:
```suggestion
			r.reqLogger.Error(err, "Error updating PagerDuty cluster config", "Name", configMapName)
```

File: pkg/controller/pagerdutyintegration/clusterdeployment_limited_support.go
Comment (jewzaam) [inline pkg/controller/pagerdutyintegration/clusterdeployment_limited_support.go:54]:
```suggestion
			r.reqLogger.Error(err, "Error enabling PagerDuty service")
```

File: pkg/controller/pagerdutyintegration/clusterdeployment_limited_support.go
Comment (jewzaam) [inline pkg/controller/pagerdutyintegration/clusterdeployment_limited_support.go:47]:
```suggestion
			r.reqLogger.Error(err, "Error updating PagerDuty cluster config", "Name", configMapName)
```

File: pkg/controller/pagerdutyintegration/clusterdeployment_limited_support.go
Comment (jewzaam) [inline pkg/controller/pagerdutyintegration/clusterdeployment_limited_support.go:40]:
```suggestion
			r.reqLogger.Error(err, "Error disabling PagerDuty service")
```

**Review by jewzaam — COMMENTED** [review]
/approve

Small nits on error logs, not a blocking thing but I do like consistency.

**Pull request merged**
